mm2src/coins/eth.rs

```diff
@@ -5478,7 +5443,9 @@ impl EthTxFeeDetails {
 impl MmCoin for EthCoin {
     fn is_asset_chain(&self) -> bool { false }
 
-    fn spawner(&self) -> CoinFutSpawner { CoinFutSpawner::new(&self.abortable_system) }
+    fn get_ctx(&self) -> Option<MmArc> { MmArc::from_weak(&self.ctx) }
```

Comment:
`ctx_from_weak` is a better name. I believe.

Reply:
IMO, `from_weak` is too much unnecessary bits of information for the caller to care about.

Reply:
> What about implementing TryFrom for MmArc so we can do eth.ctx.try_into()

What's difference between having a try from or an MmCoin method for that? This is in place in MmCoin though since we need access to the global `StreamingManager` in coins (generically).

One way to do it is to just give each coin a copy of the streaming manager (it's short, not like we gonna pass tons of channels, but i still dislike it cuz some coins already access ctx which already stores the streaming manager), or pass ctx instead which is passed in some coins already and has a streaming manager inside.
i believe every coin gonna need to access some global shared object at some point, so just doing it early on.

Reply:
> What's difference between having a try from or an MmCoin method for that? This is in place in MmCoin though since we need access to the global StreamingManager in coins (generically).

This is perfect fit for `TryFrom` trait and we don't need to bloat out coin traits with unnecessary functions.

Reply:
> What about implementing TryFrom for MmArc so we can do eth.ctx.try_into()?

Wouldn't try_into always move 'ctx' from etc?